namer/consul/src/main/scala/io/buoyant/namer/consul/CatalogNamer.scala

```diff
@@ -108,30 +122,43 @@ class CatalogNamer(
       updateableAddr() = Addr.Pending
     }
 
-    def init(): Future[Unit] = mkRequest().flatMap(update).handle(handleUnexpected)
+    private[this] def metaFuture() =
+      if (setHost)
+        domainFuture().map { domainOption =>
+          val domain = domainOption.getOrElse("consul")
+          Addr.Metadata(Metadata.authority -> s"${key.name}.service.$datacenter.$domain")
+        }
+      else
+        Future.value(Addr.Metadata.empty)
+
+    def cycle(): Future[Unit] =
+      for {
+        services <- mkRequest()
+        meta <- metaFuture()
+        result <- update(services, meta)
+      } yield cycle().handle(handleUnexpected)
```

Comment:
Can I do `yield None` and then a call to `cycle` on the next line? Or that's idiomatically incorrect?